src/DaPigGuy/libPiggyEconomy/providers/BedrockEconomyProvider.php

```diff
@@ -24,32 +24,48 @@ public static function checkDependencies(): bool
 
     public function __construct()
     {
-        $this->api = BedrockEconomyAPI::legacy();
-        $this->currency = BedrockEconomy::getInstance()->getCurrencyManager();
+        $this->api = BedrockEconomyAPI::ASYNC();
+        $this->currency = BedrockEconomy::getInstance()->getCurrency();
     }
 
     public function getMonetaryUnit(): string
     {
-        return $this->currency->getSymbol();
+        return $this->currency->symbol;
     }
 
     public function getMoney(Player $player, callable $callback): void
     {
-        $this->api->getPlayerBalance($player->getName(), ClosureContext::create(fn(?int $balance) => $callback($balance ?? $this->currency->getDefaultBalance())));
+        Await::f2c(
+            fn () => yield from $this->api->get($player->getXuid(), $player->getName()),
+            fn (array $result) => $callback($result['amount']),
+            fn () => $callback($this->currency->defaultAmount)
```

Comment:
The third parameter has been deprecated, see [here](https://github.com/SOF3/await-generator/blob/master/await-generator/src/SOFe/AwaitGenerator/Await.php#L96)